clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java

```diff
@@ -17,38 +17,20 @@
 
 package org.apache.kafka.clients.admin;
 
-import java.time.Duration;
-import java.util.Collection;
-import java.util.HashSet;
 import java.util.Map;
 import java.util.Properties;
-import java.util.Set;
-import java.util.concurrent.TimeUnit;
-import org.apache.kafka.common.ElectionType;
-import org.apache.kafka.common.Metric;
-import org.apache.kafka.common.MetricName;
-import org.apache.kafka.common.TopicPartition;
-import org.apache.kafka.common.TopicPartitionReplica;
-import org.apache.kafka.common.acl.AclBinding;
-import org.apache.kafka.common.acl.AclBindingFilter;
-import org.apache.kafka.common.annotation.InterfaceStability;
-import org.apache.kafka.common.config.ConfigResource;
 
 /**
- * The administrative client for Kafka, which supports managing and inspecting topics, brokers, configurations and ACLs.
+ * The base class for in-built admin clients.
  *
- * The minimum broker version required is 0.10.0.0. Methods with stricter requirements will specify the minimum broker
- * version required.
+ * Client code should use the newer {@link Admin} interface in preference to this class.
  *
- * This client was introduced in 0.11.0.0 and the API is still evolving. We will try to evolve the API in a compatible
- * manner, but we reserve the right to make breaking changes in minor releases, if necessary. We will update the
- * {@code InterfaceStability} annotation and this notice once the API is considered stable.
+ * This class may be removed in a later release, but has not be marked as deprecated to avoid unnecessary noise.
```

Comment:
This was probably discussed in the KIP, but obviously the downside is that users won't get any warning or hint that they should transition. But I guess we don't get a substantial benefit from removing `AdminClient`, so maybe we'll just never do it.

Reply:
I agree, but this was covered / discussed in detail on the voting thread and this was the outcome.